sympy/integrals/transforms.py

```diff
@@ -1150,9 +1156,89 @@ class LaplaceTransform(IntegralTransform):
 
     _name = 'Laplace'
 
+    def _cr(self, f, a, c, **hints):
+        conds = not hints.get('noconds', False)
+        if conds:
+            return f, a, c
+        else:
+            return f
+
     def _compute_transform(self, f, t, s, **hints):
+        a = Wild('a', exclude=[t])
+        b = Wild('b', exclude=[t])
+        c = Wild('c', exclude=[t])
+        g = WildFunction('g', nargs=1)
+        tau = Wild('tau', exclude=[t])
+        omega = Wild('omega', exclude=[t])
+        dco = lambda f: self._deep_collect(f,t)
+        if not any(func.has(t) for func in f.atoms(AppliedUndef)):
```

Comment:
Why is this `if` needed?

Reply:
Everything after the `else:`, if it triggers any rule, returns a `LaplaceTransform` and will not be evaulated further. The `if` is there because I fear infinite loops.

Reply:
I don't think I understand. Why is it checking `func.has(t)`?

Reply:
Because if it does not have `t`, it should use the normal rules.

With the present code, 

```LT(exp(-f(x)*t), t, s)```

gives 

```(1/(s + f(x)), -f(x), True)```,

processed by rule 3.1, as it should.